src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.NativeAot.cs

```diff
@@ -14,9 +16,14 @@ public partial struct GCHandle
         internal static void InternalSet(IntPtr handle, object? value) => RuntimeImports.RhHandleSet(handle, value);
 
 #if FEATURE_JAVAMARSHAL
-        // FIXME implement waiting for bridge processing
         internal static object? InternalGetBridgeWait(IntPtr handle)
         {
+            if (RuntimeImports.RhIsGCBridgeActive())
+            {
+#pragma warning disable CA1416 // Validate platform compatibility
+                JavaMarshal.WaitForGCBridgeFinish();
+#pragma warning restore CA1416 // Validate platform compatibility
+            }
             return InternalGet(handle);
```

Comment:
This needs to be moved into the native side so it's synchronized with the GC (basically so the GC can't start running the bridge after we've waited and before we get the object).